codersdk/workspaceagents.go

```diff
@@ -392,6 +392,68 @@ func (c *Client) WorkspaceAgentListeningPorts(ctx context.Context, agentID uuid.
 	return listeningPorts, json.NewDecoder(res.Body).Decode(&listeningPorts)
 }
 
+// WorkspaceAgentContainer describes a container of some sort that is visible
+// to the workspace agent.
+type WorkspaceAgentContainer struct {
```

Comment:
Open to calling this anything else, "Container" has a lot of associated baggage, whereas these are basically ephemeral resources created at runtime that we want to surface to users. Could be a Docker container, could be a Kubernetes pod, could be a `chroot`.

Reply:
I suspect you'll need different types for each, otherwise this abstraction is gonna be bloody bloody
(`chroot` has no concept of image / ports).

Reply:
I feel this could be called `WorkspaceAgentDevcontainer`, and thus be the abstraction layer between different implementations. I'd prefer we output one standardized type so we don't need to handle them in various places (UI, CLI, etc).